design/design-doc-package-refactor.md

```diff
@@ -0,0 +1,680 @@
+    apiVersion: database.gcp.crossplane.io/v1beta1
+  - kind: GKECluster
+    apiVersion: container.gcp.crossplane.io/v1beta1
+  ...
+  infrastructureDefinitions:
+  - name: fullnetwork.network.gcp.crossplane.io # provider-gcp may choose to ship infrastructure definitions to make using its services easier
+  compositions:
+  - name: bignetwork.fullnetwork.network.gcp.crossplane.io # composition that is from fullnetwork.network.gcp.crossplane.io
+  - name: smallnetwork.fullnetwork.network.gcp.crossplane.io # composition that is from fullnetwork.network.gcp.crossplane.io
+  dependsOn: [] # provider-gcp does not have any dependencies
+  title: GCP Provider
+  overviewShort: "The GCP Crossplane provider enables infrastructure management for the Google Cloud Platform."
+  overview: |
+    The Google Cloud Platform (GCP) Crossplane provider adds support for managing GCP infrastructure in Kubernetes.
+    Google Kubernetes Engine, Google Cloud SQL, Google Cloud Memorystore, and Google Cloud Storage resources can be provisioned, updated, and deleted by this provider.
+  readme: |
+    `provider-gcp` is a Crossplane infrastructure provider for the [Google Cloud Platform](https://cloud.google.com).
+    The provider that is built from the source code in this repository can be installed into a Crossplane control plane and adds the following new functionality:
+
+    * Custom Resource Definitions (CRDs) that model GCP infrastructure and services (e.g. [CloudSQL](https://cloud.google.com/sql/), [GKE clusters](https://cloud.google.com/kubernetes-engine/), etc.)
+    * Controllers to provision these resources in GCP based on the users desired state captured in CRDs they create
+    * Implementations of Crossplane's portable resource abstractions, enabling GCP resources to fulfill a user's general need for cloud services
+  category: Infrastructure
+  version: 0.1.0
+  icons:
+  - base64data: iVBORw0KGgoAAAANSUhEUgAAAOEAAADZCAYAAADWmle6AAAACXBIWXMA
+    mediatype: image/png
+  maintainers:
+  - name: Nic Cope
+   email: nic@upbound.io
+  owners:
+  - name: Nic Cope
+    email: nic@upbound.io
+  keywords:
+  - "google"
+  - "gcp"
+  - "infrastructure"
+  website: "https://crossplane.io"
+  controller:
+    deployment:
+      name: provider-gcp-controller
+      spec:
+        replicas: 1
+        selector:
+          matchLabels:
+            core.crossplane.io/name: "provider-gcp"
+        template:
+          metadata:
+            name: provider-gcp-controller
+            labels:
+              core.crossplane.io/name: "provider-gcp"
+          spec:
+            serviceAccountName: provider-gcp-controller
+            containers:
+            - name: provider-gcp-controller
+              image: crossplane/provider-gcp:v0.10.0
+              imagePullPolicy: Always
+```
+
+User creates `Package`:
+
+```yaml
+apiVersion: packages.crossplane.io/v1alpha1
+kind: Package
+metadata:
+  name: secure-stacl
+spec:
+  package: crossplane/secure-stack:v0.1.0
```

Comment:
How should the private registries be handled?

Reply:
`Package` (formerly `ClusterPackageInstall`) supports `ImagePullSecrets` https://doc.crds.dev/github.com/crossplane/crossplane/cluster/charts/crossplane-types/crds/packages.crossplane.io_clusterpackageinstalls.yaml#spec-imagePullSecrets